test/e2e-api/members/signin.test.js

```diff
@@ -0,0 +1,106 @@
+const membersService = require('../../../core/server/services/members');
+const {agentProvider, mockManager, fixtureManager} = require('../../utils/e2e-framework');
+
+let membersAgent;
+let adminAgent;
+
+describe('Members Signin', function () {
+    before(async function () {
+        // Weird - most of the mocks happen after getting the agent
+        // but to mock stripe we want to fake the stripe keys in the settings.
+        // And it's initialised at boot - so mocking it before
+        // Probably wanna replace this with a settinfs fixture mock or smth??
+        mockManager.setupStripe();
+
+        const agents = await agentProvider.getAgentsForMembers();
+        membersAgent = agents.membersAgent;
+        adminAgent = agents.adminAgent;
+
+        await fixtureManager.init('members');
+        await adminAgent.loginAsOwner();
+    });
+
+    beforeEach(function () {
+        mockManager.mockLabsEnabled('multipleProducts');
+        mockManager.mockLabsEnabled('tierWelcomePages');
+        mockManager.mockStripe();
+    });
+
+    afterEach(function () {
+        mockManager.restore();
+    });
+
+    it('Will not set a cookie if the token is invalid', async function () {
+        await membersAgent.get('/?token=blah')
+            .expectStatus(302)
+            .expectHeader('Location', /\?\w*success=false/);
+    });
+
+    it('Will set a cookie if the token is valid', async function () {
+        const magicLink = await membersService.api.getMagicLink('member1@test.com');
+        const magicLinkUrl = new URL(magicLink);
+        const token = magicLinkUrl.searchParams.get('token');
+
+        const res = await membersAgent.get(`/?token=${token}`)
+            .expectStatus(302)
+            .expectHeader('Location', /\?\w*success=true/)
+            .expectHeader('Set-Cookie', /members-ssr.*/);
+    });
+
+    it('Will redirect to the free welcome page for signup', async function () {
+        const magicLink = await membersService.api.getMagicLink('member1@test.com');
+        const magicLinkUrl = new URL(magicLink);
+        const token = magicLinkUrl.searchParams.get('token');
+
+        const {body: {products}} = await adminAgent.get('/products/');
+
+        const freeProduct = products.find(product => product.type === 'free');
+
+        await adminAgent.put(`/products/${freeProduct.id}/`).body({
```

Comment:
I don't particularly like this, but the way product fixtures work are really weird because they are fixtures in the database too, this was the quickest way to get things working.